src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -214,87 +193,189 @@ private static DataFrameColumn CreateColumn(Type kind, string[] columnNames, int
+                    indexColumn[i] = i;
                 }
-                return ret;
+                columns.Insert(0, indexColumn);
+            }
+            return ret;
+        }
+
+        private struct CsvLines : IEnumerable<string>
+        {
+            private CsvLineEnumerator enumerator;
+            public CsvLines(CsvLineEnumerator csvLineEnumerator)
+            {
+                enumerator = csvLineEnumerator;
+            }
+
+            public IEnumerator<string> GetEnumerator() => enumerator;
+
+            IEnumerator IEnumerable.GetEnumerator() => enumerator;
+        }
+
+        private struct CsvLineEnumerator : IEnumerator<string>
+        {
+            private StreamReader streamReader;
+            private string currentLine;
+            private long streamStartPosition;
+            public CsvLineEnumerator(StreamReader csvStream)
+            {
+                streamStartPosition = csvStream.BaseStream.Position;
+                streamReader = csvStream;
+                currentLine = null;
+            }
+
+            public string Current => currentLine;
+
+            object IEnumerator.Current => currentLine;
+
+            public void Dispose()
+            {
+                throw new NotImplementedException();
+            }
+
+            public bool MoveNext()
+            {
+                currentLine = streamReader.ReadLine();
+                return currentLine != null;
+            }
+
+            public void Reset()
+            {
+                streamReader.DiscardBufferedData();
+                streamReader.BaseStream.Seek(streamStartPosition, SeekOrigin.Begin);
+            }
+        }
+
+        /// <summary>
+        /// Reads CSV data passed in as a string into a DataFrame.
+        /// </summary>
+        /// <param name="csvString">csv data passed in as a string</param>
+        /// <param name="separator">column separator</param>
+        /// <param name="header">has a header or not</param>
+        /// <param name="columnNames">column names (can be empty)</param>
+        /// <param name="dataTypes">column types (can be empty)</param>
+        /// <param name="numberOfRowsToRead">number of rows to read not including the header(if present)</param>
+        /// <param name="guessRows">number of rows used to guess types</param>
+        /// <param name="addIndexColumn">add one column with the row index</param>
+        /// <param name="encoding">The character encoding. Defaults to UTF8 if not specified</param>
+        /// <returns><see cref="DataFrame"/></returns>
+        public static DataFrame LoadCsvFromString(string csvString,
+                                char separator = ',', bool header = true,
+                                string[] columnNames = null, Type[] dataTypes = null,
+                                long numberOfRowsToRead = -1, int guessRows = 10, bool addIndexColumn = false,
+                                Encoding encoding = null)
```

Comment:
Encoding doesn't make any sense for a string. strings are always UTF-16.

Reply:
Ah yup you're right. Fixed